File: src/model/structs/Move.java
package model.structs;

import model.enums.Player;
import model.pieces.Piece;

public class Move {
    public final Point start;
    public final Point end;
    public final Piece piece;
    public Move(Point start, Point end, Piece piece) {
        this.start  = new Point(start);
        this.end    = new Point(end);
        this.piece  = piece;
    }
    public Move(Point start, Point end) {
        this.start  = new Point(start);
        this.end    = new Point(end);
        this.piece  = null;
    }
}

File: src/model/structs/Move.kt
package model.structs

import model.pieces.Piece

class Move {
    val start: Point
    val end: Point
    val piece: Piece?

    constructor(start: Point, end: Point, piece: Piece?) {
        this.start = Point(start)
        this.end = Point(end)
        this.piece = piece
    }

    constructor(start: Point, end: Point) {
        this.start = Point(start)
        this.end = Point(end)
        this.piece = null
    }
}
